swaylock/main.c

```diff
@@ -100,12 +104,29 @@ bool verify_password() {
 }
 
 void notify_key(enum wl_keyboard_key_state state, xkb_keysym_t sym, uint32_t code, uint32_t codepoint) {
+	int i;
+	int redraw_screen = 0;
 	if (state == WL_KEYBOARD_KEY_STATE_PRESSED) {
 		switch (sym) {
 		case XKB_KEY_Return:
+			render_data.auth_state = AUTH_STATE_VALIDATING;
+
+			render(&render_data);
+			// Make sure our render call will actually be displayed on the screen
+			wl_display_flush(registry->display);
+
+			// However, this is not how it should be done.
+			for (i = 0; i < registry->outputs->length; ++i) {
+				if (wl_display_dispatch(registry->display) == -1) {
+					exit(0);
```

Comment:
Yeah, I think this could get stuck here with multiple outputs, if one call of `wl_display_dispatch` dispatches events for multiple windows.